ksql-parser/src/test/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercerTest.java

```diff
@@ -108,7 +111,27 @@ public void shouldNotDownCastDouble() {
     final double val = 1d;
 
     // Expect:
-    assertThat(coercer.coerce(val, SqlType.INTEGER), is(Optional.empty()));
-    assertThat(coercer.coerce(val, SqlType.BIGINT), is(Optional.empty()));
+    assertThat(coercer.coerce(val, Schema.OPTIONAL_INT32_SCHEMA), is(Optional.empty()));
+    assertThat(coercer.coerce(val, Schema.OPTIONAL_INT64_SCHEMA), is(Optional.empty()));
+  }
+
+  @Test
+  public void shouldCoerceNumberToDecimal() {
+    // Given:
+    final Object[] values = new Object[]{1, 1L, 1.0d};
+
+    // Expect:
+    for (final Object val : values) {
+      assertThat(coercer.coerce(val, DecimalUtil.builder(2, 1)), is(Optional.of(new BigDecimal("1.0"))));
+    }
+  }
+
+  @Test
+  public void shouldCoerceStringToDecimal() {
+    // Given:
+    final String val = "1.0";
+
+    // Expect:
+    assertThat(coercer.coerce(val, DecimalUtil.builder(2, 1)), is(Optional.of(new BigDecimal("1.0"))));
```

Comment:
It would be good to add tests for the failure case such as invalid string for decimal.